marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -190,16 +185,19 @@ private void applyOptionsToEndpointConstants(ObjectNode endpointConstants, Map<S
         });
     }
 
-    private void loadEndpoints() throws IOException {
-        String scriptPath = "marklogic-data-hub-spark-connector/bulkIngester.sjs";
-        String apiPath = "marklogic-data-hub-spark-connector/bulkIngester.api";
-
-        TextDocumentManager modMgr = hubClient.getModulesClient().newTextDocumentManager();
-        DocumentWriteSet writeSet = modMgr.newWriteSet();
-        DocumentMetadataHandle metadata = buildDocumentMetadata();
-        writeSet.add("/" + apiPath, metadata, new InputStreamHandle(new ClassPathResource(apiPath).getInputStream()).withFormat(Format.JSON));
-        writeSet.add("/" + scriptPath, metadata, new InputStreamHandle(new ClassPathResource(scriptPath).getInputStream()).withFormat(Format.TEXT));
-        modMgr.write(writeSet);
+    private void loadEndpointsIfNotPresent(String scriptPath, String apiPath) {
```

Comment:
To be safe, we really need to do one document at a time. If someone how module exists but the other doesn't, this will throw an error. The efficiency of writing 2 at a time isn't important here; this is going to finish in a very small amount of time no matter what, and it's a one-time operation. 

So this method should just handle a single module. It just needs to know the path and the Format.

Reply:
Done.